### PR TITLE
Checkbox styled component

### PR DESCRIPTION
**Pull request by gracesnoh**
checkbox

### CODE REVIEW

File: reflex-app/src/animations/CheckboxDemo/CheckboxDemo.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    stroke: #fff;
+  }
+`;
+
+const Checkbox = styled.div`
+  position: relative;
+  width: 36px;
+  height: 36px;
+
+  > input {
```
Comment (hidesminimally) [inline reflex-app/src/animations/CheckboxDemo/CheckboxDemo.js]: What does the `>` do in SCSS?

Comment (gracesnoh) [reply reflex-app/src/animations/CheckboxDemo/CheckboxDemo.js]: only applies to the immediate children of the parent div that are `input`

**Review by hidesminimally — APPROVED** [review]
quick review, but LGTM :shipit: 

Didn't look too closely at specifics tbh 


I would like to change this PR in the future to have two sets of animations: Checkbox, and CheckboxDemo. The idea here is that we make Checkbox easy to export to an npm package (basically removing the hover prop cause that's the only thing specific to the demo) and then implement the CheckboxDemo to use Checkbox component and add the hover prop. 

The pattern is demonstrated with the FormInput components. The Notification Bell was an old version before I thought of this pattern I'm describing. 

If you're interested in making that change, feel free to make it. It's not super important for now.

**Comment by gracesnoh** [comment]
@lilshim Makes sense - I'll take a stab at creating the Checkbox component then!

**Pull request merged**